### ISSUE TITLE
Can't get colormap to work properly

### ISSUE THREAD

**Issue by davidjurgens** [sbebo/joypy#4]
I'm trying to make a joyplot and when I set colormap=cm.autumn_r , my plot only uses the first color in the colormap.  I wanted to use cm.tab20c and I eventually got it working by hacking the `_get_color` function to return `colormap(i % len(colormap.colors))` .  I don't think this is the right general solution but I'm not sure what to fix before submitting a pull request.  Any ideas what might be going wrong?  I think there might also be some weird interaction between gradient and named-color colormaps too.

Also, this package is amazing!  Thank you so much for putting it together.

**Comment by leotac** [comment]
Hi! Could you include a minimal working example? I can't reproduce the issue.

Indeed your fix would only work with what matplotlib calls "qualitative colormaps" (see [here](https://matplotlib.org/users/colormaps.html)).
Continuous colormaps (sequential, diverging) don't have the "colors" attribute. Right now the colormap is called with a value between 0 and 1. That should work for all kinds of colormaps, continuous or not.

If the number of subplots is <= the number of available colors, that's what you should get (this is created with cm.tab20c):
![tab20](https://user-images.githubusercontent.com/1311290/29753041-7286ec78-8b69-11e7-8704-87484af2e79d.png)

The only issue I see with the current approach is that you might get repeated colors with qualitative colormaps that have a _smaller_ number of colors with respect to the number of subplots. E.g., with cm.tab10, you can get something like this:
![repeated](https://user-images.githubusercontent.com/1311290/29752998-b6443a0c-8b68-11e7-88d0-8b877cf7e017.png)
which is probably undesirable. But it doesn't seem to be your problem.

**Comment by dfiorino** [comment]
Hello, 

I ran into the same problem while trying to run on **Python v2.7.13**. I posted it to Stack Overflow and was presented a work-around (https://stackoverflow.com/questions/46717394/demo-of-joypy-joyplots-in-python-not-working). It's the addition of `from __future__ import division` to the top of `joyplot.py`. Obviously, the intention of this project isn't to support every version of Python, but this hack allows me to use this awesome package!

<img width="870" alt="screen shot 2017-10-13 at 11 38 40 am" src="https://user-images.githubusercontent.com/23364185/31554301-219b73e4-b00b-11e7-8819-634676e8496c.png">
